core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -18,541 +18,562 @@
+  }
+
+  /**
+   * Loads, or reloads, the topic stored by this service given a
+   * specified topic ID. See setTopic() for more information on
+   * additional behavior of this function.
+   */
+  loadTopic(topicId: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this._topicIsLoading = true;
+      this.editableTopicBackendApiService.fetchTopic(
+        topicId).then(
+        (newBackendTopicObject) => {
+          this._skillCreationIsAllowed = (
             newBackendTopicObject.skillCreationIsAllowed);
-          _skillQuestionCountDict = (
+          this._skillQuestionCountDict = (
             newBackendTopicObject.skillQuestionCountDict);
-          _updateGroupedSkillSummaries(
+          this._updateGroupedSkillSummaries(
             newBackendTopicObject.groupedSkillSummaries);
-          _updateTopic(
+          this._updateTopic(
             newBackendTopicObject.topicDict,
             newBackendTopicObject.skillIdToDescriptionDict
           );
-          _updateGroupedSkillSummaries(
+          this._updateGroupedSkillSummaries(
             newBackendTopicObject.groupedSkillSummaries);
-          _updateSkillIdToRubricsObject(
+          this._updateSkillIdToRubricsObject(
             newBackendTopicObject.skillIdToRubricsDict);
-          _updateClassroomUrlFragment(
+          this._updateClassroomUrlFragment(
             newBackendTopicObject.classroomUrlFragment);
-          _updateTopicRights(newBackendTopicRightsObject);
-          _setCanonicalStorySummaries(canonicalStorySummaries);
-          _topicIsLoading = false;
-          $rootScope.$applyAsync();
-        }, (error) => {
-          AlertsService.addWarning(
-            error || 'There was an error when loading the topic editor.');
-          _topicIsLoading = false;
+          this.editableTopicBackendApiService.fetchStories(topicId).then(
+            (canonicalStorySummaries) => {
+              this._setCanonicalStorySummaries(canonicalStorySummaries);
+            });
+        },
+        (error) => {
+          this.alertService.addWarning(
+            error || 'There was an error when loading the topic.');
+          this._topicIsLoading = false;
+          reject();
         });
-      },
+      this.topicRightsBackendApiService.fetchTopicRights(
+        topicId).then((newBackendTopicRightsObject) => {
+        this._updateTopicRights(newBackendTopicRightsObject);
+        this._topicIsLoading = false;
+        resolve();
+      }, (error) => {
+        this.alertService.addWarning(
+          error ||
+            'There was an error when loading the topic rights.');
+        this._topicIsLoading = false;
+        reject();
+      });
+    });
+  }
 
-      getGroupedSkillSummaries: function() {
-        return angular.copy(_groupedSkillSummaries);
-      },
+  getGroupedSkillSummaries(): unknown {
+    return cloneDeep(this._groupedSkillSummaries);
```

Comment:
clone deep is used to created a separate object, I don't think so the return type would be unknown here.

Reply:
yes done